src/kyber_py/ml_kem/ml_kem.py

```diff
@@ -173,21 +173,29 @@ def _pke_keygen(self):
     def _pke_encrypt(self, ek_pke, m, r):
         """
         Algorithm 13
+
+        As well as performing the usual pke encryption, the FIPS document
+        requires two additional checks.
+
+        1. Type Check: The ek_pke is of the expected length
+        2. Modulus Check: That t_hat has been canonically encoded
         """
+        # These should always hold, as encaps() generates m, r and
+        # _pke_encrypt should never be called directly by a user
         assert len(m) == 32
         assert len(r) == 32
 
+        # First check if the encap key has the right length
+        if len(ek_pke) != 384 * self.k + 32:
+            raise ValueError("Type check failed, ek_pke has the wrong length")
```

Comment:
would be nice to user to error out with "expected X bytes, gotten Y bytes"...

Reply:
I addressed this in #75